src/components/HomePage/SearchResultsArea/SearchResultsWidget/SearchResultsMobile.js

```diff
@@ -128,8 +158,12 @@ const SearchResultsMobile = () => {
                 <OfferViewer
                     open={openPreview}
                     toggleOpenPreview={toggleOpenPreview}
-                    offerContentWrapperClassName={classes.offerBodyContainer}
-                    selectedOffer={selectedOffer}
+                    offerWidgetWrapperClassName={classes.offerBodyContainer}
+                    offer={selectedOfferIdx !== null && offers?.length > 0 ? offers[selectedOfferIdx] : null}
```

Comment:
Add a condition for `offers?.length > selectedOfferIdx` so that imagining that we have selected the last offer, and redo the search with tighter restrictions (fewer results) it won't crash since offers[selectedIdx] might not exist

Reply:
It should not happen due to the loading state setting the idx to null, but still, it's a nice safeguard for the future (filters being applied might limit the number of offers without triggering the null of the idx)